### PR TITLE
System Cleanup

### PR DESCRIPTION
**Pull request by mashrukk**
## System Cleanup Script
This is a comprehensive system cleanup script that performs various cleanup tasks.
#### Key Features:
- Removes unused packages
- Common cleanup tasks (journalctl, temporary files, log truncation)
- Optional user cache and trash cleanup
This script provides a unified approach to system cleanup across different Linux distributions, improving system performance and freeing up disk space.
## Type of Change
- [x] New feature
## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/system-setup/system-cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            $ESCALATION_TOOL "$PACKAGER" tidy
+            $ESCALATION_TOOL "$PACKAGER" cc -a
+            ;;
+        dnf)
+            $ESCALATION_TOOL "$PACKAGER" clean all
+            $ESCALATION_TOOL "$PACKAGER" autoremove -y
+            ;;
+        pacman)
+            $ESCALATION_TOOL "$PACKAGER" -Sc --noconfirm
+            $ESCALATION_TOOL "$PACKAGER" -Rns "$(pacman -Qtdq)" --noconfirm
```
Comment (adamperkowski) [inline tabs/system-setup/system-cleanup.sh:26]:
```suggestion
            "$ESCALATION_TOOL" "$PACKAGER" clean
            "$ESCALATION_TOOL" "$PACKAGER" autoremove -y
            "$ESCALATION_TOOL" "$PACKAGER" autoclean
            "$ESCALATION_TOOL" du -h /var/cache/apt
            "$ESCALATION_TOOL" "$PACKAGER" clean
            ;;
        zypper)
            "$ESCALATION_TOOL" "$PACKAGER" clean -a
            "$ESCALATION_TOOL" "$PACKAGER" tidy
            "$ESCALATION_TOOL" "$PACKAGER" cc -a
            ;;
        dnf)
            "$ESCALATION_TOOL" "$PACKAGER" clean all
            "$ESCALATION_TOOL" "$PACKAGER" autoremove -y
            ;;
        pacman)
            "$ESCALATION_TOOL" "$PACKAGER" -Sc --noconfirm
            "$ESCALATION_TOOL" "$PACKAGER" -Rns "$(pacman -Qtdq)" --noconfirm
```

File: tabs/system-setup/system-cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            ;;
+        *)
+            printf "%b\n" "${RED}Unsupported package manager: $PACKAGER${RC}"
+            return 1
+            ;;
+    esac
+
+    printf "%b\n" "${GREEN}System cleanup completed.${RC}"
```
Comment (adamperkowski) [inline tabs/system-setup/system-cleanup.sh:34]: Remove this line or change it to
```suggestion
    printf "%b\n" "${GREEN}Package manager cleanup completed.${RC}"
```

File: tabs/system-setup/system-cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+common_cleanup() {
+    $ESCALATION_TOOL find /var/tmp -type f -atime +5 -delete
+    $ESCALATION_TOOL find /tmp -type f -atime +5 -delete
+    $ESCALATION_TOOL find /var/log -type f -name "*.log" -exec truncate -s 0 {} \;
+    $ESCALATION_TOOL journalctl --vacuum-time=3d
```
Comment (adamperkowski) [inline tabs/system-setup/system-cleanup.sh:41]:
```suggestion
    "$ESCALATION_TOOL" find /var/tmp -type f -atime +5 -delete
    "$ESCALATION_TOOL" find /tmp -type f -atime +5 -delete
    "$ESCALATION_TOOL" find /var/log -type f -name "*.log" -exec truncate -s 0 {} \;
    "$ESCALATION_TOOL" journalctl --vacuum-time=3d
```

File: tabs/system-setup/system-cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+clean_data() {
+    printf "%b\n" "${YELLOW}Clean up old cache files and empty the trash? (y/N): ${RC}"
+    read -r clean_response
+    case $clean_response in
+        [Yy]*)
```
Comment (adamperkowski) [inline tabs/system-setup/system-cleanup.sh:48]:
```suggestion
        y|Y)
```

To match other scripts.

**Review by adamperkowski — CHANGES_REQUESTED** [review]

File: tabs/system-setup/system-cleanup.sh (hunk 1 above)
Comment (mashrukk) [reply tabs/system-setup/system-cleanup.sh:26]: Should it be within qoutes? I removed them cause i thought $ESCALATION_TOOL is correct

Comment (adamperkowski) [reply tabs/system-setup/system-cleanup.sh:26]: Should be. #454 would get merged today, but Chris messed it up when trying to resolve conflicts.

**Comment by nnyyxxxx** [comment]
@mashrukk Can you stop spamming PRs? I saw your history and you just keep closing and then re-opening the PR for some reason? I assume a bump to the top, please don't do that. It's getting pretty redundant...
image

**Comment by mashrukk** [comment]
@nnyyxxxx My bad. ok I'll try

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**